labkey-client-api/build.gradle

```diff
@@ -57,20 +57,19 @@ repositories {
 
 group "org.labkey.api"
 
-version "2.1.0-SNAPSHOT"
+version "2.1.0-httpclient-SNAPSHOT"
 
 dependencies {
-    implementation "org.apache.httpcomponents:httpmime:${httpmimeVersion}"
     api ("com.googlecode.json-simple:json-simple:${jsonSimpleVersion}")
             {
                 // exclude this because it gets in the way of our own JSON object implementations from server/api
                 exclude group: "org.json", module:"json"
             }
+    api "org.apache.httpcomponents.client5:httpclient5:${httpclient5Version}"
+    api "org.apache.httpcomponents.core5:httpcore5:${httpcore5Version}"
```

Comment:
Just checking that these are actually api dependencies (that is, we expose some objects from these jar files in our interfaces).  I assume so since they were before.

Reply:
Yes, unfortunately our wrappers expose a few native `HttpClient`/`HttpCore` classes. For example, our `Connection.executeRequest()` returns `CloseableHttpResponse` (from HttpClient) which implements `ClassicHttpResponse` (from HttpCore). I've made a few internal methods private in this PR to reduce the leakage, but we can't eliminate it without more work.